server/db/context/users_context.ts

```diff
@@ -188,6 +188,66 @@ export const getUserById = async (userId: number) => {
 	}
 };
 
+export const getUsersInfo = async ({
+	index,
+	perPage,
+	nickname,
+	email,
+}: GetUsersInfoParams) => {
+	let conn: PoolConnection | null = null;
+	try {
+		let sql = `SELECT COUNT(*) OVER() as total ,u.id, u.email, u.nickname, u.created_at, u.isDelete,
+		(SELECT COUNT(id) FROM comments WHERE author_id = u.id) as comment_count,
+		(SELECT COUNT(id) FROM posts WHERE author_id = u.id) as post_count 
+		FROM users u 
+		`;
+
+		const value = [];
+
+		if (nickname || email) {
+			sql += `WHERE `;
+		}
+
+		if (nickname) {
+			sql += `u.nickname LIKE ? `;
+			value.push(`%${nickname}%`);
+
+			if (email) {
+				sql += `AND `;
+			}
+		}
+
+		if (email) {
+			sql += `u.email LIKE ? `;
+			value.push(`%${email}%`);
+		}
+
+		sql += `ORDER BY u.id ASC`;
+
+		const pagenationSQL = ` LIMIT ? OFFSET ?`;
+
+		sql += pagenationSQL;
+		value.push(perPage);
+		value.push(index * perPage);
+
+		conn = await pool.getConnection();
+		const [rows]: [IUserInfoRow[], FieldPacket[]] = await conn.query(
+			sql,
+			value
+		);
```

Comment:
@Siltarae 가독성과 유지보수 측면에서 코멘트 드립니다.

WHERE 조건이 많고 복잡할때 Query Building 시 가독성을 향상시킬 수 있는 방법이 있는데요,
filters, values 배열로 조건을 분리하면 가독성도 좋아지고 유지보수하기 편리해질 것 같습니다.

중복된 if 분기문도 제거할 수 있을 것 같아요.
```suggestion
	try {
		const filters: string[] = [];
		const values: (string | number)[] = [];

		if (nickname) {
			filters.push(`u.nickname LIKE ?`);
			values.push(`%${nickname}%`);
		}

		if (email) {
			filters.push(`u.email LIKE ?`);
			values.push(`%${email}%`);
		}

		let sql = `
			SELECT COUNT(*) OVER() as total, u.id, u.email, u.nickname, u.created_at, u.isDelete,
			(SELECT COUNT(id) FROM comments WHERE author_id = u.id) as comment_count,
			(SELECT COUNT(id) FROM posts WHERE author_id = u.id) as post_count
			FROM users u
		`;

		if (filters.length > 0) {
			sql += ` WHERE ` + filters.join(' AND ');
		}

		sql += ` ORDER BY u.id ASC LIMIT ? OFFSET ?`;
		values.push(perPage, index * perPage);

		conn = await pool.getConnection();
		const [rows]: [IUserInfoRow[], FieldPacket[]] = await conn.query(
			sql,
			values
		);
```